mm2src/coins/solana/solana_tests.rs

```diff
@@ -124,6 +125,84 @@ mod tests {
         assert_eq!(res.is_valid, false);
     }
 
+    #[tokio::test]
+    #[cfg(not(target_arch = "wasm32"))]
+    async fn solana_transaction_simulations() {
+        let passphrase = "federal stay trigger hour exist success game vapor become comfort action phone bright ill target wild nasty crumble dune close rare fabric hen iron".to_string();
+        let (_, sol_coin) = solana_coin_for_test(passphrase.clone(), SolanaNet::Devnet);
+        let valid_tx_details = sol_coin
+            .withdraw(WithdrawRequest {
+                coin: "SOL".to_string(),
+                from: None,
+                to: sol_coin.my_address.clone(),
+                amount: BigDecimal::from(0.0001),
+                max: false,
+                fee: None,
+            })
+            .compat()
+            .await
+            .unwrap();
+        assert_eq!(valid_tx_details.spent_by_me, BigDecimal::from_str("0.000105").unwrap());
+        assert_eq!(
+            valid_tx_details.received_by_me,
+            BigDecimal::from_str("0.000095").unwrap()
+        );
+        assert_eq!(valid_tx_details.total_amount, BigDecimal::from_str("0.000105").unwrap());
```

Comment:
If I'm correct, `total_amount` is expected to be the same as `WithdrawRequest::amount`, i.e `0.0001`.
Please look at the ETH [implementation](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/eth.rs#L643)